src/FuncSharp/Collections/IEnumerableExtensions_ReadOnlyList.cs

```diff
@@ -1,100 +1,122 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
+using System.Diagnostics.Contracts;
 
 namespace FuncSharp;
 
 public static partial class IEnumerableExtensions
 {
-    public static bool NonEmpty<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool NonEmpty<T>(this IReadOnlyCollection<T> collection)
     {
-        return source is not null && source.Count > 0;
+        return collection is not null && collection.Count > 0;
     }
 
-    public static bool IsEmpty<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool IsEmpty<T>(this IReadOnlyCollection<T> collection)
     {
-        return !source.NonEmpty();
+        return !collection.NonEmpty();
     }
 
-    public static bool IsMultiple<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool IsMultiple<T>(this IReadOnlyCollection<T> collection)
     {
-        return source is not null && source.Count > 1;
+        return collection is not null && collection.Count > 1;
     }
 
-    public static bool IsSingle<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool IsSingle<T>(this IReadOnlyCollection<T> collection)
     {
-        return source is not null && source.Count == 1;
+        return collection is not null && collection.Count == 1;
```

Comment:
Evaluating against constant can be safely done with `is` operator, even though I don't expect anyone to override the == 😅 
```suggestion
        return collection?.Count is 1;
```

Reply:
Based on our 🤙🏻 we'll keep it as it is